src/main/java/com/arpnetworking/http/Routes.java

```diff
@@ -104,7 +103,11 @@ public Routes(
      */
     public Flow<HttpRequest, HttpResponse, NotUsed> flow() {
         return Flow.<HttpRequest>create()
-                .mapAsync(1, this);
+                .mapAsync(1, this)
+                .recover(
+                        new PFBuilder<Throwable, HttpResponse>()
+                                .match(Exception.class, e -> HttpResponse.create().withStatus(StatusCodes.INTERNAL_SERVER_ERROR))
```

Comment:
So on what was line 224, there is already an exceptionally in the flow which _may_ map everything to 404. I also don't see where parse exception is caught and mapped to 400. I'm not convinced this works in all cases.